### PR TITLE
Provide a game topic extractor

### PR DESCRIPTION
**Pull request by john-peterson**
This provides a game topic extractor from within newpipe so you dont have to go out to the slow sluggy ugly horrendoues browser every time. asnd keep pictures turned off and everything and just find a quiert game stream with no voewers and no hype

i have a pounding headache now but i wilkl make an apk and pusgh the last pr to the other repo.

### CODE REVIEW

File: extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java [hunk 1]
```diff
@@ -538,6 +541,38 @@ public String getUploaderUrl() throws ParsingException {
...
 
+    @Override
+    public String getTopicUrl() throws ParsingException {
+        // This is not in the api yet topicDetails is empty
+        // https://developers.google.com/youtube/v3/docs/videos/list
+        String url = getUrl();
+        try {
+            String data = "";
+            Document doc = Jsoup.connect(url).get();
+            org.jsoup.select.Elements el = doc.select("script");
```
Comment (Stypox) [inline extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java:552]: You should not make another request. Surely the topic url information is stored in one of the `JsonObject` fields of this class (probably `nextResponse`). You can probably replace `json` below with `nextResponse`.

File: extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java [hunk 2]
```diff
@@ -538,6 +541,38 @@ public String getUploaderUrl() throws ParsingException {
         throw new ParsingException("Could not get uploader url");
     }
 
+    @Override
+    public String getTopicUrl() throws ParsingException {
+        // This is not in the api yet topicDetails is empty
+        // https://developers.google.com/youtube/v3/docs/videos/list
+        String url = getUrl();
```
Comment (Stypox) [inline extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java:548]: What is this? Why is this variable also used for returning?

File: extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java [hunk 3]
```diff
@@ -538,6 +541,38 @@ public String getUploaderUrl() throws ParsingException {
...
+            org.jsoup.select.Elements el = doc.select("script");
+            for (org.jsoup.nodes.Element e : el) {
+                if (e.data().contains("ytInitialData")) {
+                    data = e.data();
+                    break;
+                }
+            }
+            String answer = data.substring(data.indexOf("{"), data.length()-1);
+            JsonObject json = JsonParser.object().from(answer);
+            url = json.getObject("contents")
```
Comment (Stypox) [inline extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeStreamExtractor.java:561]: Just do `return` here and remove the `return` below

**Review by Stypox — CHANGES_REQUESTED** [review]

**Comment by AudricV** [comment]
Closing due to lack of author's response.

**Pull request closed without merging**